### PR TITLE
PUB-735: Build Subscriptions URN Search Screen

### PR DESCRIPTION
**Pull request by marcogagliardi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:
image

image

image

### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-735

### Change description ###
Added urn search page and result page

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    });
+  });
+});
+
+describe(`getSubscriptionUrnDetails(${invalidUrn})`, () => {
+
+  stub.withArgs(invalidUrn).returns({});
+
+
+  it('should return empty list as urn ${invalidCourtId}', () => {
```
Comment (junaidiqbalmoj) [inline src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:58]: It shouldn't be invalidCourtId.

File: src/test/unit/controllers/SubscriptionUrnSearchController.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+const api = new PipApi(axios);
+const subscriptionUrnSearchController = new SubscriptionUrnSearchController(api);
+const stub = sinon.stub(api, 'getSubscriptionByUrn');
+
+describe('Subscription Urn Search Controller', () => {
+  it('should render the search page', () => {
+
+
+    const rawData = fs.readFileSync(path.resolve(__dirname, '../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
+    const hearingsData = JSON.parse(rawData);
```
Comment (junaidiqbalmoj) [inline src/test/unit/controllers/SubscriptionUrnSearchController.test.ts:21]: change variable name hearingsData to something else. i.e. subscriptionResult

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts (hunk 1 above)
Comment (marcogagliardi) [reply src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:58]: good spot. corrected

File: src/test/unit/controllers/SubscriptionUrnSearchController.test.ts (hunk 2 above)
Comment (marcogagliardi) [reply src/test/unit/controllers/SubscriptionUrnSearchController.test.ts:21]: yes I agree. corrected

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const stub = sinon.stub(api, 'getSubscriptionByUrn');
+const rawData = fs.readFileSync(path.resolve(__dirname, '../../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
+const subscriptionsData = JSON.parse(rawData);
+
+describe(`getSubscriptionUrnDetails(${validUrn})`, () => {
+
+  stub.withArgs(validUrn).returns(subscriptionsData);
+
+
+  it('should return list of hearings', () => {
```
Comment (junaidiqbalmoj) [inline src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:27]: Should it be hearings or cases?

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  it('should have mocked object in the subscriptions list', () => {
+    return subscriptionActions.getSubscriptionUrnDetails(validUrn).then(data => {
+      expect(data.filter((hearings) => hearings.urn === data[0].urn).length).toBe(1);
+    });
+  });
+
+  it(`should have only subscriptions for urn ${validUrn}`, () => {
+    return subscriptionActions.getSubscriptionUrnDetails(validUrn).then(data => {
+      expect(data.filter((hearings) => hearings.urn === validUrn).length).toBe(data.length);
```
Comment (junaidiqbalmoj) [inline src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:48]: Should it be hearing or case?

File: src/test/unit/views/subscription-search-urn-results.test.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const searchTerm = '123456789';
+const numOfResults = '1';
+const PAGE_URL = `/subscription-search-urn-results?search-input=${searchTerm}`;
+
+const rowClass = 'govuk-table__row';
+
+let htmlRes: Document;
+
+const rawData = fs.readFileSync(path.resolve(__dirname, '../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
+const hearingsData = JSON.parse(rawData);
```
Comment (junaidiqbalmoj) [inline src/test/unit/views/subscription-search-urn-results.test.ts:17]: change hearingData variable name.

File: src/test/unit/views/subscription-urn-search.test.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+const errorSummaryBodyClass = 'govuk-error-summary__body';
+const formErrorClass = 'govuk-form-group--error';
+
+const expectedHeader = 'Enter a unique reference number';
+const expectedButtonText = 'Continue';
+
+let htmlRes: Document;
+
+const rawData = fs.readFileSync(path.resolve(__dirname, '../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
+const hearingsData = JSON.parse(rawData);
```
Comment (junaidiqbalmoj) [inline src/test/unit/views/subscription-urn-search.test.ts:23]: change hearingsData variable name

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts (hunk 3 above)
Comment (marcogagliardi) [reply src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:27]: corrected

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts (hunk 4 above)
Comment (marcogagliardi) [reply src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:48]: corrected

File: src/test/unit/views/subscription-search-urn-results.test.ts (hunk 5 above)
Comment (marcogagliardi) [reply src/test/unit/views/subscription-search-urn-results.test.ts:17]: corrected

File: src/test/unit/views/subscription-urn-search.test.ts (hunk 6 above)
Comment (marcogagliardi) [reply src/test/unit/views/subscription-urn-search.test.ts:23]: corrected

File: src/test/unit/controllers/SubscriptionSearchUrnResultController.test.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+const api = new PipApi(axios);
+const subscriptionSearchUrnResultController = new SubscriptionSearchUrnResultController(api);
+const stub = sinon.stub(api, 'getSubscriptionByUrn');
+
+describe('Subscription Search Urn Result Controller', () => {
+  it('should render the urn search result page', () => {
+
+
+    const rawData = fs.readFileSync(path.resolve(__dirname, '../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
+    const hearingsData = JSON.parse(rawData);
```
Comment (junaidiqbalmoj) [inline src/test/unit/controllers/SubscriptionSearchUrnResultController.test.ts:21]: need to change variable name here as well.

Comment (marcogagliardi) [reply src/test/unit/controllers/SubscriptionSearchUrnResultController.test.ts:21]: done

File: src/main/views/subscription-search-urn-results.njk [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+{% block pageTitle %}
+  {{ "Subscription URN Search Results" | safe }}
+{% endblock %}
+
+{% block beforeContent %}
+  {{ goBack('/subscription-urn-search') }}
+{% endblock %}
+
+{% block content %}
+  <h1 class="govuk-heading-m">Search result</h1>
```
Comment (ChrisS1512) [inline src/main/views/subscription-search-urn-results.njk:14]: I think the search result heading is slightly smaller than the screenshot

File: src/main/routes/routes.ts [hunk 9]
```diff
@@ -23,6 +23,10 @@ export default function(app: Application): void {
   app.get('/otp-login', app.locals.container.cradle.otpLoginController.get);
   app.get('/otp-login-testing', cors(corsOptions), app.locals.container.cradle.otpLoginTestingController.get);
   app.post('/otp-login', app.locals.container.cradle.otpLoginController.post);
+  app.get('/subscription-urn-search', app.locals.container.cradle.subscriptionUrnSearchController.get);
+  app.post('/subscription-urn-search', app.locals.container.cradle.subscriptionUrnSearchController.post);
+  app.get('/subscription-search-urn-results', app.locals.container.cradle.subscriptionSearchUrnResultController.get);
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/routes/routes.ts:28]: corrected

File: src/test/unit/controllers/SubscriptionUrnSearchController.test.ts [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+    responseMock.expects('render').once().withArgs('subscription-urn-search');
+
+    return subscriptionUrnSearchController.post(request, response).then(() => {
+      responseMock.verify();
+    });
+  });
+
+  it('should redirect to urn search results page with input as query if urn input is valid', () => {
+    const rawData = fs.readFileSync(path.resolve(__dirname, '../../../main/resources/mocks/subscriptionListResult.json'), 'utf-8');
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/unit/controllers/SubscriptionUrnSearchController.test.ts]: I think we can do this in a separate task. we need to change all other tests as well

File: src/main/views/subscription-urn-search.njk [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+{% block pageTitle %}
+  {{ "Search for a URN" | safe }}
+{% endblock %}
+
+{% block beforeContent %}
+  {{ goBack('/') }}
+{% endblock %}
+
+{% block content %}
+  {% if noResultsError or invalidInputError %}
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search.njk:15]: corrected. Sorry I missed it from the task description

File: src/main/views/subscription-urn-search.njk [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+          },
+          id: "search-input",
+          name: "search-input",
+          autocomplete: "false"
+        }) }}
+      </div>
+
+    </div>
+    {{ submitButton() }}
+    {% if noResultsError %}
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search.njk:61]: I thought it was better to be consistent with the other errors

File: src/test/routes/subscriptionUrnSearch.test.ts [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import { expect } from 'chai';
+import request from 'supertest';
+
+import { app } from '../../main/app';
+
+describe('subscription URN Search', () => {
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/routes/subscriptionUrnSearch.test.ts:6]: added

File: src/main/resources/mocks/subscriptionListResult.json [hunk 14]
```diff
@@ -0,0 +1,9 @@
+[
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/resources/mocks/subscriptionListResult.json:1]: this is used in the mock data for the LocalApi

File: src/test/unit/controllers/SubscriptionUrnSearchController.test.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/unit/controllers/SubscriptionUrnSearchController.test.ts]: POST test added

File: src/main/views/subscription-search-urn-results.njk (hunk 8 above)
Comment (marcogagliardi) [reply src/main/views/subscription-search-urn-results.njk]: corrected

File: src/main/views/subscription-urn-search.njk [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+    </div>
+  {% endif %}
+  {% set inputClasses = 'govuk-div govuk-input--error'
+    if noResultsError or invalidInputError else 'govuk-!-width-two-thirds' %}
+  {% set formGroupError = 'govuk-form-group--error'
+    if noResultsError or invalidInputError else 'govuk-form-group' %}
+  <form action="subscription-urn-search" method="post">
+    <div class="govuk-form-group govuk-!-width-one-half">
+      <h1 class="govuk-label-wrapper">
+        <label class="govuk-label govuk-label--m" for="search-input">
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search.njk]: corrected

**Review by ChrisS1512 — APPROVED** [review]

**Review by junaidiqbalmoj — CHANGES_REQUESTED** [review]
Please address the comments I put on the files.

File: src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts [hunk 16]
```diff
@@ -0,0 +1,33 @@
+import { HearingListPage } from './HearingList.page';
+
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts:2]: removed

File: src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+  async getResults(): Promise<number> {
+    $(helpers.Results).catch(() => {
+      console.log(`${helpers.Results} not found`);
+    });
+
+    const results = $$(helpers.Results);
+    return results.length;
+  }
+
+  async selectCourt(): Promise<HearingListPage> {
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts:24]: removed

File: src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    $(helpers.LinkResult).catch(() => {
+      console.log(`${helpers.LinkResult} not found`);
+    });
+    const linkResult = await $(helpers.LinkResult);
+    linkResult.click();
+
+    return new HearingListPage();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/e2e/pageobjects/SubscriptionUrnSearchResults.page.ts:31]: removed

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  it('should return list of 1 cases', () => {
+    return subscriptionActions.getSubscriptionUrnDetails(validUrn).then(data => {
+      expect(data.length).toBe(1);
+    });
+  });
+
+
+  it('should have mocked object in the cases list', () => {
+    return subscriptionActions.getSubscriptionUrnDetails(validUrn).then(data => {
+      expect(data.filter((hearings) => hearings.urn === data[0].urn).length).toBe(1);
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:42]: name changed

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+    });
+  });
+
+  it(`should have only cases for urn ${validUrn}`, () => {
+    return subscriptionActions.getSubscriptionUrnDetails(validUrn).then(data => {
+      expect(data.filter((hearings) => hearings.urn === validUrn).length).toBe(data.length);
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:48]: name changed

File: src/main/views/subscription-urn-search.njk [hunk 21]
```diff
@@ -0,0 +1,70 @@
+{% extends "template.njk" %}
+{% from "macros/common-components.njk" import goBack, submitButton %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+
+{% block pageTitle %}
+  {{ "Search for a URN" | safe }}
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search.njk:7]: title changed

File: src/main/resources/actions/subscriptionSearchActions.ts [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+
+
+  public async getSubscriptionUrnDetails(urn): Promise<Array<Subscription>> {
+    const subscriptions = await this.api.getSubscriptionByUrn(urn);
+
+    if (subscriptions) {
+      return subscriptions;
+    } else {
+      console.log(`Subscription with urn ${urn} does not exist`);
+      return null;
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/resources/actions/subscriptionSearchActions.ts:16]: changed to empty array

File: src/test/unit/controllers/SubscriptionUrnSearchController.test.ts [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+    const subscriptionResult = JSON.parse(rawData);
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/unit/controllers/SubscriptionUrnSearchController.test.ts]: lines moved

**Review by junaidiqbalmoj — APPROVED** [review]

File: src/main/views/subscription-urn-search.njk [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+{% from "macros/common-components.njk" import goBack, submitButton %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+
+{% block pageTitle %}
+  {{ "Search case by URN" | safe }}
+{% endblock %}
+
+{% block beforeContent %}
+  {{ goBack('/') }}
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search.njk:11]: done

File: src/main/views/subscription-urn-search-results.njk [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+    <tr class="govuk-table__row">
+      <th scope="col" class="govuk-table__header">Unique reference number</th>
+      <th scope="col" class="govuk-table__header">Case reference number</th>
+      <th scope="col" class="govuk-table__header">Case name</th>
+    </tr>
+    </thead>
+    <tbody class="govuk-table__body">
+    {% for result in searchResults %}
+      <tr class="govuk-table__row">
+        <th scope="row" class="govuk-table__cell--no-line">{{ result.urn }}</th>
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search-results.njk:28]: done

File: src/main/controllers/SubscriptionUrnSearchController.ts [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+    res.render('subscription-urn-search');
+  }
+
+
+  public async post(req: Request, res: Response): Promise<void> {
+    const searchInput = req.body['search-input'];
+
+    if (searchInput && searchInput.length >= 3) {
+      const searchResults = await new SubscriptionSearchActions(_api).getSubscriptionUrnDetails(searchInput);
+      (searchResults && searchResults.length > 0) ?
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/controllers/SubscriptionUrnSearchController.ts:22]: done

File: src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+      expect(data.filter((subcription) => subcription.urn === validUrn).length).toBe(data.length);
+    });
+  });
+});
+
+describe(`getSubscriptionUrnDetails(${invalidUrn})`, () => {
+
+  stub.withArgs(invalidUrn).returns({});
+
+
```
Comment (kurt-uk) [inline src/test/unit/actions/subscriptionSearchActions/subscriptionSearchActions.test.ts:57]: just the 1 blank line is what we go for

File: src/test/unit/controllers/SubscriptionUrnSearchResultController.test.ts [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+
+
+const api = new PipApi(axios);
+const subscriptionSearchUrnResultController = new SubscriptionUrnSearchResultController(api);
+const stub = sinon.stub(api, 'getSubscriptionByUrn');
+
+describe('Subscription Search Urn Result Controller', () => {
+  it('should render the urn search result page', () => {
+
+
```
Comment (kurt-uk) [inline src/test/unit/controllers/SubscriptionUrnSearchResultController.test.ts:19]: again lots of white space

File: src/main/views/subscription-urn-search.njk [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+    if noResultsError or invalidInputError else 'govuk-form-group' %}
+  <form action="subscription-urn-search" method="post">
+    <div class="govuk-form-group govuk-!-width-one-half">
+      <h1 class="govuk-label-wrapper">
+        <label class="govuk-label govuk-label--l" for="search-input">
+          Enter a unique reference number
+        </label>
+      </h1>
+      <div class="{{ formGroupError }}">
+        <div id="search-input-hint" class="govuk-hint">
```
[the remark this answers is not in the document]
Comment (commitBlob) [reply src/main/views/subscription-urn-search.njk:50]: Just to explain the reasoning behind this comment.
If you look at the line 50 `"text": ''`  you are setting this to blank to satisfy accessibility issues right? What I am trying to suggest here is to remove lines 43-45 and add `Please provide the exact match` to the label text.
This way you will reduce file by 3 lines and satisfy accessibility.
From the visual perspective it doesn't change anything, but will probably be more useful to visually impaired people

File: src/test/e2e/tests/specs.test.ts [hunk 30]
```diff
@@ -177,6 +187,35 @@ describe('Finding a court or tribunal listing', () => {
     });
   });
 
+  describe('Following the subscription \'search\' path', () => {
+    const validSearchTerm = '123456789';
+    const invalidSearchTerm = '12345678';
+    const expectedNumOfResults = 1;
+
+    it('should open URN search page with Enter a unique reference number', async () => {
+      await subscriptionUrnSearchPage.open('/subscription-urn-search');
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/test/e2e/tests/specs.test.ts:196]: what does it mean?

File: src/main/resources/mocks/subscriptionList.json [hunk 31]
```diff
@@ -0,0 +1,30 @@
+[
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply src/main/resources/mocks/subscriptionList.json:1]: json file changed

File: src/main/models/subscription.ts [hunk 32]
```diff
@@ -0,0 +1,11 @@
+export interface Subscription {
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/models/subscription.ts:1]: i didnt mean any offence in my comment? im sorry you took that instruction to back your opinion with data offensively, the model is supposed to reflect how the data is managed, so with missing fields you dont capture all the data

File: src/main/controllers/SubscriptionUrnSearchController.ts [hunk 33]
```diff
@@ -0,0 +1,30 @@
...
+
+  public get(req: Request, res: Response): void {
+    res.render('subscription-urn-search');
+  }
+
+
+  public async post(req: Request, res: Response): Promise<void> {
+    const searchInput = req.body['search-input'];
+
+    if (searchInput && searchInput.length >= 3) {
```
Comment (junaidiqbalmoj) [inline src/main/controllers/SubscriptionUrnSearchController.ts:20]: Is it required to check the length?

File: src/main/views/subscription-urn-search-results.njk [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+{% block content %}
+  <h1 class="govuk-heading-l">Search result</h1>
+  <h3 class="govuk-heading-s">{{ searchResults.length }} result successfully found</h3>
+
+  <table class="govuk-table govuk-!-width-two-thirds">
+    <thead class="govuk-table__head">
+    <tr class="govuk-table__row">
+      <th scope="col" class="govuk-table__header">Unique reference number</th>
+      <th scope="col" class="govuk-table__header">Case reference number</th>
+      <th scope="col" class="govuk-table__header">Case name</th>
```
Comment (junaidiqbalmoj) [inline src/main/views/subscription-urn-search-results.njk:29]: It should be right aligned.

File: src/main/views/subscription-urn-search-results.njk [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+    </tr>
+    </thead>
+    <tbody class="govuk-table__body">
+    {% for result in searchResults %}
+      <tr class="govuk-table__row">
+        <td scope="row" class="govuk-table__cell--no-line">{{ result.urn }}</td>
+        <td scope="row" class="govuk-table__cell--no-line">{{ result.caseNumber }}</td>
+        <td class="govuk-table__cell--no-line">{{ result.caseName }}</td>
```
Comment (junaidiqbalmoj) [inline src/main/views/subscription-urn-search-results.njk:37]: It should be right aligned.

File: src/main/views/subscription-urn-search.njk [hunk 36]
```diff
@@ -0,0 +1,74 @@
...
+  {% endif %}
+  {% set inputClasses = 'govuk-div govuk-input--error'
+    if noResultsError or invalidInputError else 'govuk-!-width-two-thirds' %}
+  {% set formGroupError = 'govuk-form-group--error'
+    if noResultsError or invalidInputError else 'govuk-form-group' %}
+  <form action="subscription-urn-search" method="post">
+    <div class="govuk-form-group govuk-!-width-one-half">
+      <h1 class="govuk-label-wrapper">
+        <label class="govuk-label govuk-label--l" for="search-input">
+          Enter a unique reference number
```
Comment (junaidiqbalmoj) [inline src/main/views/subscription-urn-search.njk:46]: Can we have the heading in one line?

File: src/main/views/subscription-urn-search-results.njk [hunk 37]
```diff
@@ -0,0 +1,43 @@
...
+{% endblock %}
+
+{% block beforeContent %}
+  {{ govukBackLink({
+    text: text,
+    href: '#',
+    attributes: {
+      onClick: "history.back();event.preventDefault();"
+    }
+  }) }}
```
Comment (junaidiqbalmoj) [inline src/main/views/subscription-urn-search-results.njk:17]: When I search for a URN, it showed me the result page. But on this page, when I click on Back button, it showed an error. It is happening on Chrome, but on Safari, it is working fine.

**Review by junaidiqbalmoj — CHANGES_REQUESTED** [review]

File: src/main/views/subscription-urn-search-results.njk (hunk 37 above)
Comment (marcogagliardi) [reply src/main/views/subscription-urn-search-results.njk:17]: I can't replicate this. I can click on the back button and it is working correctly in Chrome, Safari and Firefox

**Comment by johnoneill46** [comment]
> Looks good :)
> 
> My only thought is that this currently won't work in staging / dev environments as we don't have a subscriptions URN endpoint on business rules (or soon to be data management).
> 
> @johnoneill46 - Are we OK with a separate ticket being created to link this up to the backend (once the other changes are done), or do we feel the backend bit also needs to be done as part of this?

Fine to be done as separate ticket PUB-735 is just for the actual UI screen - logic can be done separately

**Review by junaidiqbalmoj — APPROVED** [review]

**Comment by commitBlob** [comment]
Closing PR as there is another pending one including this work (#128 )
Re-open if necessary

**Pull request closed without merging**